src/transaction.js

```diff
@@ -257,7 +258,12 @@ const _states = {
         })
         .catch(error => new FailedObserver({ error, onError }))
 
-      return newCompletedResult(observerPromise, query, parameters)
+      return newCompletedResult(
+        observerPromise,
+        query,
+        parameters,
+        connectionHolder
```

Comment:
Wouldn't it be easier to initialize completed result with protocol version instead of connection holder? I mean the transaction should have used a valid connection at some point (like _begin should have been called succesfully) and wherever that happens the protocol version could be stashed away and reused here.

Reply:
Yes, it could be simpler. I will check if i have it unwrapped outside a promise some where on the session.run and on the transaction.run

Reply:
In the moment the result is created, we only have a promise of connection, not the connection.

Reply:
Haven't _begin executed? I guess it must execute before run somehow..

Reply:
Yes, but in the Session.run, we don't have it unwrapped. We could inform a promise of protocol version.